src/node_buffer.cc

```diff
@@ -701,6 +928,50 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void IsNull(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!HasInstance(args[0]))
+    return env->ThrowTypeError("first arg should be a Buffer");
+
+  ARGS_THIS(args[0].As<Object>())
```

Comment:
Technically all C++ operations should be doing a `HasInstance()` check. Mainly because the `instanceof` check in JS isn't sufficient to prevent this from aborting:
```js
function Foo() { }
Foo.prototype.__proto__ = Buffer.prototype;
var f = new Foo();
f.slice(0, 1);
```